src/pugixml.hpp

```diff
@@ -122,6 +122,26 @@
 #	endif
 #endif
 
+// Detect if C++ standard supports 'string_view' (2017 or higher)
```

Comment:
I would propose simplifying this to enabling internal PUGI_HAS_STRING_VIEW define if `PUGIXML_STRING_VIEW` define (note the `PUGIXML` prefix; `PUGI_` prefix is for internal macros, `PUGIXML_` prefix is for user facing configuration) is enabled, and dropping the `_SUPPORTED` macro as it does not seem to be useful.

Separately, `PUGI_HAS_STRING_VIEW` should be undef'd at the end of the header.

`PUGIXML_STRING_VIEW` should be added, commented out, in `pugiconfig.hpp`. We can note there that the macro will become unnecessary with future versions.

Reply:
A question regarding consistency: `PUGI_HAS_STRING_VIEW` seems to be the only macro using `PUGI` prefix *in the header*. All others use `PUGIXML`, e.g., `PUGIXML_HAS_MOVE` or `PUGIXML_HAS_LONG_LONG` which are auto-detected as well. `PUGI`-macros can only be found in the cpp.
So shouldn't it be `PUGIXML_HAS_STRING_VIEW`?

Probably for the same reason it would be the only macro that is actually undefined in the header. (there is no `#undef` to be found) So I cannot orient on anything. Should I always `#undef` it even if I didn't defined it? (not that there is a `#ifndef PUGI_HAS_STRING_VIEW`)

Reply:
Apologies, I've misled you about the prefix; I forgot the conventions :) Indeed, it should be prefixed with `PUGIXML_` and no undef is necessary. I mixed this up with the `.cpp` file where the prefix is `PUGI_IMPL_` and the macros are undef'd to not pollute the namespace in header-only mode.